### ISSUE TITLE
Bots - Refilling loaders objective progress resets when interrupted by flooding

### ISSUE THREAD

**Issue by Kaillera** [Regalis11/Barotrauma#3752]
- [ ] I have searched the issue tracker to check if the issue has already been reported.

**Description**
When the bot is going to refill a loader, and all of a sudden a flood happens in the path inbetween when coming back with ammo boxes, the bot goes to get a diving suit and returns to the periscope with ammo box in hand, instead of completing the objective of refilling the loader. Then it remembers to refill the loader, but may end up running back to a shelf instead of using the one on hand.

**Version**
~0.10.502

**Comment by itchyOwl** [comment]
Yeah, that's the way the objectives work for now. Resetting the objectives when they are selected may cause these issues, but not resetting them causes other issues. It's still something I should try solving at some point, if there's time.
